crates/core/tedge/src/cli/mqtt/cli.rs

```diff
@@ -61,6 +62,25 @@ impl BuildCommand for TEdgeMqttCli {
             .load()?
             .query(MqttClientCapathSetting)
             .ok();
+        let client_cert = context
+            .config_repository
+            .load()?
+            .query(MqttClientAuthCertSetting);
+        let client_private_key = context
+            .config_repository
+            .load()?
+            .query(MqttClientAuthKeySetting);
+
+        let client_auth_config = if client_cert.is_err() && client_private_key.is_err() {
+            None
+        } else {
```

Comment:
What the logic here? To ignore the config if both are incorrect?

Reply:
Yes, if neither `client_cert` nor `client_key` are present, then we skip client authentication

Reply:
In see, one has to read `is_err()` as `is_none()`, but this is misleading as we have a `ConfigSettingError`. 

This is okay for now as there is another [on-going work on tedge-config](https://github.com/thin-edge/thin-edge.io/issues/1812). However, is see here yet another opportunity to improve  error management as you did with https://github.com/thin-edge/thin-edge.io/pull/1874: the same error type is used both for get and set.